pkgs/initiator/initiator.go

```diff
@@ -102,13 +108,24 @@ type Payload struct {
 	SharesData  string   `json:"sharesData"`  // encrypted with RSA keys private BLS shares of each operator participating in DKG
 }
 
+type pongResult struct {
+	ip     string
+	err    error
+	result []byte
+}
+
 // GeneratePayload generates at initiator ssv smart contract payload using DKG result  received from operators participating in DKG ceremony
-func GeneratePayload(result []dkg.Result, sigOwnerNonce []byte) (*KeyShares, error) {
+func GeneratePayload(result []dkg.Result, sigOwnerNonce []byte, owner common.Address, nonce uint64) (*KeyShares, error) {
 	// order the results by operatorID
 	sort.SliceStable(result, func(i, j int) bool {
 		return result[i].OperatorID < result[j].OperatorID
 	})
-
+	sorted := sort.SliceIsSorted(result, func(p, q int) bool {
+		return result[p].OperatorID < result[q].OperatorID
+	})
+	if !sorted {
+		return nil, fmt.Errorf("slice isnt sorted")
```

Comment:
not very informative error, which slice?

Reply:
fixed